File: data-structures/Array/One-Dimensional-Array/Basic Operations/array.js
'use simple'

// This is about basic use of array in JS 
// It covers all the built-in functions that support arrays

// Declaration
// let num = new Array();
// let num = [];
let num = [19, 10, 8, 17, 9];

//traverse through the array and print all the elements
for (let index = 0; index < num.length; index++) {
    alert(`num[${index}] is equal to ${val}`, index, num[index]);
}

//update element at i index
num[3] = -20;
alert(`num[3] is ${num[3]}`, num[3]);

//access out of bound
alert(`num[3] is ${num[5]}`, num[5]);

//get array size
alert("array size is "+ num.length);

File: data-structures/Array/One-Dimensional-Array/Basic Operations/array.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

//main method
int main(){
    //variables
    int num[5] = {19, 10, 8, 17, 9};

    //print all the elements in the array
    //traversing through the array
    for (int i = 0; i < 5; i++)
    {
        printf("num[%d] =  %d", i, num[i]);
    }
    
    //update element at position i
    num[3] = -20;
    printf("num[%d] is %d", 3, num[3]);

    //accessing out of bound element
    printf("num[5] is %d", num[5]);

    return 1;
}